client/src/components/UserAuthMenu.vue

```diff
@@ -0,0 +1,146 @@
+                      flat
+                      icon="mdi-cog"
+                    ></v-btn>
+                  </template>
+              </v-list-item>
+              <v-list-item>
+                <v-switch
+                  @change="updateLocalLoginData()" 
+                  v-model="KeepLoggedIn"
+                  color="purple"
+                  label="Stay Logged In"
+                  ></v-switch>
+              </v-list-item>
+
+            </v-list>
+            <v-divider></v-divider>
+            <v-card-actions>
+              <v-btn
+                @click="logOutHandler" 
+                variant="text"
+                prepend-icon="mdi-logout"
+              >Log Out</v-btn>
+              <v-spacer></v-spacer>
+              <v-btn
+                variant="text"
+                @click="showProfileMenu = false"
+              >Close</v-btn>
+            </v-card-actions>
+          </v-card>
+        </v-menu>
+  <!-- diff -->
+    
+</template>
+<script setup lang="ts">
+  import { ref, computed, onMounted} from "vue"
+  import { useDisplay } from 'vuetify'
+  import { useGoogleUserData } from '../stores/useDataStore'
+  import { useDialog } from '../stores/useUserInterfaceStore'
+  const { getGoogleUserData, logUserOut, logUserIn} = useGoogleUserData()
+  const { showDialog } = useDialog()
+  const { smAndDown, mdAndUp, smAndUp, xs } = useDisplay();
+  
+  const googleProfileData = computed(() => {
+    return getGoogleUserData()
+  })
+  const showProfileMenu = ref(false)
+
+  const KeepLoggedIn = ref<any>() // unsure what the switch stores.
+
+  const saveLoginDataLocally = () => {
+  try {
+    localStorage.setItem('googleProfileData', JSON.stringify(googleProfileData.value))
+  } catch (error) {
+      console.error('Error writing variable to localStorage:', error)
+  }
+}
+
+  const checkIfKeepLoggedIn = () => {
+    try {
+      const googleProfileData = localStorage.getItem('googleProfileData')
+      return googleProfileData !== null
+    } catch (error) {
+      console.error('Error retrieving variable from localStorage:', error)
+      return false;
+    }
+  }
+
+  const loadLocalLoginData = () => {
+  try {
+    const localLoginData = localStorage.getItem('googleProfileData')
+    if (localLoginData) return JSON.parse(localLoginData)
+    else return getGoogleUserData()
+
+  } catch (error) {
+    console.error('Error retrieving variable from localStorage:', error)
+    return getGoogleUserData()
+  }
+}
+
+  const keepUserLoggedIn = ref(checkIfKeepLoggedIn())
+
+  onMounted(() => {
+    if (keepUserLoggedIn.value) logUserIn(loadLocalLoginData())
+  })
+
+  const updateLocalLoginData = () => {
+  if (keepUserLoggedIn.value) saveLoginDataLocally()
+  else deleteLocalLoginData()
+
+  }
+
+  const logOutHandler = () =>{
+    showProfileMenu.value = false;
+    deleteLocalLoginData();
+    logUserOut();
+  }
+
+  const deleteLocalLoginData = () => {
```

Comment:
this is now in two places. I honestly don't know if its worth separating this menu. If you want to move the logic into a separate function file you can, however I don't necessarily see the benefit to moving it out. Logins are now handled in two places instead of one.

Reply:
actually your best bet might be to just keep all the login stuff together, and just pass the functions in as props if you really want this to be a separate component.